django/core/files/move.py

```diff
@@ -85,8 +85,9 @@ def file_move_safe(
     except PermissionError as e:
         # Certain filesystems (e.g. CIFS) fail to copy the file's metadata if
         # the type of the destination filesystem isn't the same as the source
-        # filesystem; ignore that.
-        if e.errno != errno.EPERM:
+        # filesystem. This also happens with SELinux-enabled systems. Ignore
+        # that.
+        if e.errno not in {errno.EACCES, errno.EPERM}:
```

Comment:
[`PermissionError`](https://docs.python.org/3/library/exceptions.html#PermissionError) only corresponds to `errno.EACCES`, `errno.EPERM` is there a reason to inspect the `errno`? :thinking:

Reply:
Is it safe to ignore all `PermissionError`s?

Reply:
Nice catch! Indeed there is no reason to check errno.  
Thinking about safety of this code I beleave it's safer to use `copymode` before `copystat`. Although `copymode` may fail with cifs it should work with SELinux in case relabel is disallowed.  
Let me take a pause and test those use-cases.

Reply:
I tested SELinux and CIFS with `file_move_safe` and here my observations:

1. I was unable to reproduce issue #28170 with current Samba and CentOS 7 client. From the backtrace in #28170 it looks like the call to `utime` is failing with EPERM, however it works for me.
2. In any case when `copystat` is failing and we ignore it then permissions doesn't got moved to new file. We must do it.
3. I beleave `copymode` must not fail with CIFS setups like in #28170, but I not 100% sure.
4. I also tested `mv` from `coreutils` and it's moving files with SELinux, prints warning about inability to set selinux permissions and exits with success.
    ![image](https://user-images.githubusercontent.com/3389586/190652030-97c6925f-729d-4487-abb3-a8f99c4ac3a5.png)

P.S.: In Py 3.11 there is new errno introduced in [PermissionError](https://docs.python.org/3.11/library/exceptions.html#PermissionError) - `ENOTCAPABLE` which is ok to ignore as I think.
P.S.S.: There is also opinion that current shutil behavior with permission errors is [Python bug](https://github.com/python/cpython/issues/42948), I agree with it.

Reply:
@djackson-saa and @peterfarrell, Can I ask for your opinions?